cookie-store/resources/cookie_helper.py

```diff
@@ -18,52 +18,76 @@
 #   Used to set cookies from an HTTP response header context.
 #
 # The response has 200 status and content-type: text/plain; charset=<charset>
-import cgi, encodings, os, re, sys, urllib
+import encodings, re
+
+from six import binary_type, PY3
+
+from six.moves.urllib.parse import parse_qs, quote
+
+from wptserve.utils import isomorphic_decode, isomorphic_encode
 
 # NOTE: These are intentionally very lax to permit testing
-DISALLOWED_IN_COOKIE_NAME_RE = re.compile(r'[;\0-\x1f\x7f]');
-DISALLOWED_IN_HEADER_RE = re.compile(r'[\0-\x1f\x7f]');
+DISALLOWED_IN_COOKIE_NAME_RE = re.compile(br'[;\0-\x1f\x7f]')
+DISALLOWED_IN_HEADER_RE = re.compile(br'[\0-\x1f\x7f]')
 
 # Ensure common charset names do not end up with different
 # capitalization or punctuation
 CHARSET_OVERRIDES = {
     encodings.codecs.lookup(charset).name: charset
-    for charset in ('utf-8', 'iso-8859-1',)
+    for charset in (u'utf-8', u'iso-8859-1',)
 }
 
+def quote_str(cookie_str):
+  if PY3:
+    if isinstance(cookie_str, binary_type):
+      cookie_str = isomorphic_decode(cookie_str)
+    return quote(cookie_str, '', encoding='iso-8859-1')
+  else:
+    return quote(cookie_str, '')
+
+def parse_qs_str(query_str):
+  if PY3:
+    if isinstance(query_str, binary_type):
+      query_str = isomorphic_decode(query_str)
+    return parse_qs(query_str, keep_blank_values=True, encoding='iso-8859-1')
+  else:
+    return parse_qs(query_str, keep_blank_values=True)
+
 def main(request, response):
   assert request.method in (
-      'GET',
-      'POST',
-  ), 'request method was neither GET nor POST: %r' % request.method
-  qd = (request.url.split('#')[0].split('?', 1) + [''])[1]
-  if request.method == 'POST':
-    qd += '&' + request.body
-  args = cgi.parse_qs(qd, keep_blank_values = True)
-  charset = encodings.codecs.lookup(args.get('charset', ['utf-8'])[-1]).name
+      u'GET',
+      u'POST',
+  ), u'request method was neither GET nor POST: %r' % request.method
+  qd = (isomorphic_encode(request.url).split(b'#')[0].split(b'?', 1) + [b''])[1]
+  if request.method == u'POST':
+    qd += b'&' + request.body
+  args = parse_qs_str(qd)
+
+  charset = encodings.codecs.lookup(args.get(u'charset', [u'utf-8'])[-1]).name
   charset = CHARSET_OVERRIDES.get(charset, charset)
-  headers = [('content-type', 'text/plain; charset=' + charset)]
+  headers = [(b'content-type', b'text/plain; charset=' + isomorphic_encode(charset))]
   body = []
-  if request.method == 'POST':
+  if request.method == u'POST':
     for set_cookie in args.get('set-cookie', []):
       if '=' in set_cookie.split(';', 1)[0]:
         name, rest = set_cookie.split('=', 1)
```

Comment:
Missing prefixes here?

Reply:
Prefixes were deliberately left out here. I think `set_cookie` here is a native string here. Is it okay?

Reply:
Ahh right, because it's from `parse_qs`...

Unfortunately, this will be problematic if/when we start to enforce prefixes for all string literals. I think we'd need to add some logic to both `quote_str` and `parse_qs_str` to always return binary strings (add `isomorphic_encode` in the PY3 branch).

Reply:
Thank @Hexcles! Code updated. Please review.